src/backend/mpi/mpi.cpp

```diff
@@ -352,13 +353,13 @@ namespace argo {
 			 */
 			void _fetch_add(global_ptr<void> obj, void* value,
 					MPI_Datatype t_type, void* output_buffer) {
-				sem_wait(&ibsem);
 				// Perform the exchange operation
-				MPI_Win_lock(MPI_LOCK_EXCLUSIVE, obj.node(), 0, globalDataWindow[0]);
-				MPI_Fetch_and_op(value, output_buffer, t_type, obj.node(), obj.offset(), MPI_SUM, globalDataWindow[0]);
-				MPI_Win_unlock(obj.node(), globalDataWindow[0]);
-				// Cleanup
-				sem_post(&ibsem);
+				std::size_t win_index = get_data_win_index(obj.offset());
```

Comment:
This code block is very repetitive.  Should it be factored out in some macro or inline function?